src/main/java/com/shurjomukhi/Shurjopay.java

```diff
@@ -291,54 +226,43 @@ public PaymentReq getDefaultInfo(PaymentReq paymentReq) {
 		
 		try {
 			paymentReq.setClientIp(InetAddress.getLocalHost().getHostAddress());
-			
 		} catch (UnknownHostException e) {
-			
 			log.warn("Client IP does not found. Setting default ip address..", e);
-			paymentReq.setClientIp(DEFAULT_IP);
 		}
 		
 		return paymentReq;
 	}
 
 	/**
-	 * Sends request to shurjopay.
-	 * @param httpBody String JSON body.
-	 * @param endPoint shurjopay endpoint
-	 * @return HttpRequest successful http request.
-	 */
-	private HttpRequest postRequest(String httpBody, String endPoint) {
-		
-		return HttpRequest.newBuilder(URI.create(spConfig.getApiBaseUrl().concat(endPoint)))
-						  .POST(HttpRequest.BodyPublishers.ofString(httpBody)).header("Content-Type", "application/json").build();
-	}
-
-	/**
-	 * Sends request to shurjopay.
+	 * Sends request to shurjopay
+	 *
 	 * @param httpBody String JSON body.
-	 * @param endPoint shurjopay endpoint
+	 * @param url request URL
+	 * @param isAuthHead true if needed "Authorization" header otherwise false
 	 * @return HttpRequest successful http request.
 	 */
-	private HttpRequest postRequest(String httpBody, String endPoint, boolean isAuthHead) {
-		
-		return HttpRequest.newBuilder(URI.create(spConfig.getApiBaseUrl().concat(endPoint)))
-						  .header(AUTH_KEYWORD, getFormattedToken(authToken.getToken(), authToken.getTokenType()))
-						  .POST(HttpRequest.BodyPublishers.ofString(httpBody)).header("Content-Type", "application/json").build();
+	private HttpRequest postRequest(String httpBody, String url, boolean isAuthHead) {
+		var request = HttpRequest.newBuilder(URI.create(url));

```

Comment:
private HttpRequest postRequest(String httpBody, String url, boolean isAuthHead) {
		var builder = HttpRequest.newBuilder(URI.create(url))
				.POST(HttpRequest.BodyPublishers.ofString(httpBody))
				.header("Content-Type", "application/json");
		if (isAuthHead) builder.header("Authorization", authToken.getFormatted());
		return builder.build();
	}

Reply:
applied this changes